### PR TITLE
Herd

### PR DESCRIPTION
**Pull request by midsum-salrux**
Dependency management.

Sample herdfile:

```
:~
  [%our %base [%da ~2022.9.5]]
  [%our %landscape [%da ~2022.9.5]]
  [%our %garden [%da ~2022.9.5]]
==
```

Build with `-citadel!commit [%foo-dev %foo]`

https://urbit.org/grants/herd

### CODE REVIEW

File: desk/mar/herd.hoon [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    =/  [who=$@(%our ship) des=desk ver=$@(%trak case)]  beef
+    ;:  weld
+      "  ["
+      ?+  who  (trip (scot %p who))
+        %our  "%our"
+      ==  " "
+      "%"  (trip des)  " "
+      (spit-ver ver)
+      "]\0a"
+    ==
```
Comment (ryjm) [inline desk/mar/herd.hoon:30]:
```suggestion
    =-  "[{-} {<des>} {(spit-ver ver)}]\0a"
    ?:  =(%our who)  <who>  <(scot %p who)>
```

Comment (ryjm) [reply desk/mar/herd.hoon:30]: interpolation is more fun ;)

File: desk/ted/commit.hoon [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/-  spider
+/-  *herd
+/+  sio=strandio
```
Comment (ryjm) [inline desk/ted/commit.hoon:3]: either import as `*strandio` or `=strandio` - both are pretty standard across `%base` so it's a little confusing when they're aliased differently.

File: desk/ted/commit.hoon [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    !<  (unit [@tas @tas])  arg
+  =/  [dev=@tas build=@tas]  (need arguments)
+  ;<  our=@p  bind:m  get-our:sio
+  ;<  ~  bind:m
+    %:  poke:sio  [our %hood]
+      [%kiln-fuse !>([build ~])]
+    ==
+  ;<  ~  bind:m
+    (poke:sio [our %hood] [%kiln-commit !>([dev %.n])])
+  ;<  ~  bind:m  (sleep:sio ~s2)
```
Comment (ryjm) [inline desk/ted/commit.hoon:21]: what's this for? if we need it, possible to make the wait time shorter?

File: desk/desk.herd [hunk 4]
```diff
@@ -0,0 +1,3 @@
+:~
+  [%our %base [%da ~2022.9.5]]
```
[the remark this answers is not in the document]
Comment (ryjm) [reply desk/desk.herd:2]: (or we can the above thread/cli command create that example)

File: desk/ted/commit.hoon [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  =/  arguments
+    !<  (unit [@tas @tas])  arg
+  =/  [dev=@tas build=@tas]  (need arguments)
+  ;<  our=@p  bind:m  get-our:sio
+  ;<  ~  bind:m
+    %:  poke:sio  [our %hood]
+      [%kiln-fuse !>([build ~])]
+    ==
+  ;<  ~  bind:m
+    (poke:sio [our %hood] [%kiln-commit !>([dev %.n])])
```
[the remark this answers is not in the document]
Comment (ryjm) [reply desk/ted/commit.hoon:20]: (or just both in wide form)

File: desk/ted/commit.hoon (hunk 3 above)
Comment (midsum-salrux) [reply desk/ted/commit.hoon:21]: Without the sleep, I ran into some issues where the commit wasn't finished when the fuse started. One second is probably enough though.

File: desk/app/herd.hoon [hunk 6]
```diff
@@ -12,11 +12,31 @@
   `this
 ++  on-save   on-save:def
 ++  on-load   on-load:def
-++  on-poke   on-poke:def
+++  on-poke
+  ::  https://developers.urbit.org/reference/arvo/clay/tasks#next---await-next
+  ::  TODO receive poke to start watching a desk
```
Comment (ryjm) [inline desk/app/herd.hoon]: p.s look up `outshine-mode` for emacs, it works really nicely for hoon files

File: desk/app/citadel.hoon [hunk 7]
```diff
@@ -572,6 +572,7 @@
       %cli  valet
       %hel  crier
       %prt  [(fuse press foundation) ~]
+      %hrd  rancher
```
Comment (ryjm) [inline desk/app/citadel.hoon:575]: perfect name

**Review by ryjm — APPROVED** [review]
LGTM

**Pull request merged**